sdp/processors/modify_manifest/data_to_data.py

```diff
@@ -13,15 +13,218 @@
+    the duration_field will be updated with the value -1.0.
+
+    Args:
+        audio_file_key (str): Key to get path to wav file.
+        duration_key (str): Key to put to audio duration.
+    Returns:
+        All the same fields as in the input manifest plus duration_field
+    """
+
+    def __init__(
+        self,
+        audio_file_key: str,
+        duration_key: str,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.audio_file_key = audio_file_key
+        self.duration_key = duration_key
+
+    def process_dataset_entry(self, data_entry):
+        audio_filepath = data_entry[self.audio_file_key]
+        try:
+            data, samplerate = sf.read(audio_filepath)
+            data_entry[self.duration_key] = data.shape[0] / samplerate
+        except Exception as e:
+            logger.warning(str(e) + " file: " + audio_filepath)
+            data_entry[self.duration_key] = -1.0
+        return [DataEntry(data=data_entry)]
+
+
+class FfmpegConvert(BaseParallelProcessor):
+    """
+    Processor for converting video or audio files to audio using FFmpeg and updating the dataset with the path to the resampled audio.
+    If id_key is not None it is used as an output file name. If id_key is None the output file name is the same as input file name with different extension
+    and input file name saves to id_key back.
+    Args:
+        resampled_audio_dir (str): The directory to store the resampled audio files.
+        media_file_key (str): The field in the dataset representing the path to the input video or audio files and store the path to the resampled audio files with .wav extension.
+        id_key (str): The field in the dataset representing the unique ID or identifier for each entry. Defaults to None.
+        target_samplerate (int, optional): The target sampling rate for the resampled audio. Defaults to 16000.
+        target_nchannels (int, optional): The target number of channels for the resampled audio. Defaults to 1.
+        **kwargs: Additional keyword arguments to be passed to the base class `BaseParallelProcessor`.
+
+    """
+
+    def __init__(
+        self,
+        resampled_audio_dir: str,
+        media_file_key: str,
+        id_key: str = None,
+        target_samplerate: int = 16000,
+        target_nchannels: int = 1,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.media_file_key = media_file_key
+        self.id_key = id_key
+        self.resampled_audio_dir = resampled_audio_dir
+        self.target_samplerate = target_samplerate
+        self.target_nchannels = target_nchannels
+
+    def prepare(self):
+        os.makedirs(self.resampled_audio_dir, exist_ok=True)
+
+    def process_dataset_entry(self, data_entry):
+        input_file = data_entry[self.media_file_key]
+        if self.id_key:
+            key = data_entry[self.id_key]
+            os.makedirs(os.path.join(self.resampled_audio_dir, key.split("/")[0]), exist_ok=True)
```

Comment:
```suggestion
            os.makedirs(os.path.join(self.resampled_audio_dir, *key.split("/")[:-1]), exist_ok=True)
```

Reply:
It took me a bit of time to understand why it says `key.split("/")`, not just `key`.
Then I realized it's like this to allow key to be `new_dir_name/new_filename`, right?

My code suggestion will allow us to generalize to `new_dir_name1/new_dir_name2/.../new_filename`. What do you think? Note that I have not tested this on any files, if you could please test that would be great.